crypto/evp/legacy_sha.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/opensslconf.h>
+
+# include <openssl/obj_mac.h>
+# include "crypto/evp.h"
+
+static const EVP_MD sha1_md = {
+    NID_sha1,
+    NID_sha1WithRSAEncryption,
```

Comment:
Won't this cause things like EVP_MD_size to suddenly start failing with implicit digests, e.g.

`int size = EVP_MD_size(EVP_sha1());`

Reply:
Oh gosh darn it, you're right...

On the other hand, it would be bad if, say due to a found bug, a caller ends up getting one answer here, but a different answer with a fetched (explicit or explicit) implementation of the same.  ...  that leads to me wondering if these kinds of functions shouldn't do an implicit fetch, but I cannot that thought makes me happy.

Reply:
I'd suggest keeping the `md_size`, `flags` and `block_size` members of the structure in your cut down implementation.

Reply:
What are the ramifications of these calling fetch?

- Could fail but then the algorithm isn't available for use, although
- No library context
- Slower

Reply:
I don't particularly see a need for them to call fetch for these fixed values. We just keep the fixed values in the cut-down structure.

Reply:
Keep the fixed values...

Reply:
Got it.  I'm disgusted by the info duplication, but yeah

Reply:
That could be our way of doing the caching (for blocksize for example) ? i.e have a pointer to the const object that we retrieve the constants from??

Reply:
Nooooooooooooo!
Please, let's stop dispersing algorithm information all over the place

Reply:
Sorry for the outburst.  What I meant to say is that the information that belongs with the provided implementations should stay with them.  Let's not add more dependencies on libcrypto nuggets.

Reply:
mmmm nuggets.. tasty..

Reply:
... except when they are aged and moldy :wink:

Reply:
Hmmmm, week old green chicken.  Always eat your greens......